### ISSUE TITLE
SR-18-0000132 FD Re-assign Elevators and create Scheduled inspections

### ISSUE THREAD

**Issue by RikiSjachrani-COM** [City-of-Madison/Development-Team-2#30]
FD Re-assign Elevators and create Scheduled inspections - 
"Attached is the completed elevator spreadsheet to schedule Annual Inspections for all of our records. A few caveats:
•	If the record already has a scheduled Annual Inspection (under our new workflow) – don’t change it
•	If it does NOT have a scheduled Annual Inspection – schedule one for the date listed
•	Nothing else in the file changes except the Annual Inspection getting scheduled
These are broken up into three separate sheets (one per inspector). Please assign all of the current elevators assigned to Shannon Drezek to the Unassigned Inspector placeholder if you could.
Also, please activate the reports needed to print inspection and reinspection worksheets as well for the Elevator folks."

\\\Fps1\data1\Information Technology\Applications and Systems\Accela\FireInspections\Change Requests\Elevator Guide sheet and Workflow Process revisit\SR-18-0000132 FD Re-assign Elev and Sch Insp

 - [ ] 1. Re-assign the Elevator Record Assigned To from the excel sheet to the new inspector [BPERMIT]
 - [ ] 3. Update any scheduled inspection with a inspection type of "Annual" to "Annual Inspection" [G6ACTION]
- [ ] 2. Update any scheduled inspection from the Assigned To or excel spreadsheet [G6ACTION]
- [ ] 4. Update the "Annual Inspection" Scheduled is in the correct month from the spreadsheet and update it if it's not [G6ACTION]
- [ ] 5. Insert a "Annual Inspection" if it doesn't have a scheduled inspection from the spreadsheet [G6ACTION]

Develop initial queries in Dev, and use the replication database for closer results. Then let's go over this before moving to production. Use Transaction rollbacks to ensure quality

**Comment by BrendenPautsch-COM** [comment]
Riki,   I have created 4 of the 5 queries and I am finalizing the Insert New Inspections.  I created the queries looking at the RptRepl data source, but I will test them against Dev early next week when I get back in.  I hope to have the queries ready for review by Tuesday.

- Brenden

**Comment by RikiSjachrani-COM** [comment]
Where are the queries at? I would like to take a look at them

**Comment by BrendenPautsch-COM** [comment]
I have them in my SR folder on my personal folder.

The main file is the 18-0000132 where all the queries except the insert the new inspections, that I am not done with yet.  I tried to note what each section of code does.  Also since the RptRepl data base is on ELAMDBTEST, I used the MadsionAccela DB on that instance fir the temp table  (zzElevatorTemp).  My plan is once I get the queries ready to test and apply to the data, I was going to move from TEST to DEV.

Brenden

**Comment by BrendenPautsch-COM** [comment]
Riki,  I have attached the SQL File.  I have run it as is with the transaction rollback against DEV and placed my notes in the header.  Please review.  I want to have this reviewed before I run against the DEV without rollback to make sure I everything covered because making any updates my be difficult and hard to verify.

Also, the only question that I have is when I update the Inspection from Annual to Annual Inspections, do you only want the current or all the unperformed inspections?

Brenden

[SR-18-000132.txt](https://github.com/City-of-Madison/Development-Team-2/files/2001857/SR-18-000132.txt)

**Comment by BrendenPautsch-COM** [comment]
Ok, Let me know, if not I can change it.

**Comment by BrendenPautsch-COM** [comment]
Riki,
Can you follow up with FIRE and see how they want to deal with 431 Permits assigned to UW, Keep them as assigned?

Here are some of the stats:
![fd stats](https://user-images.githubusercontent.com/30356949/40129899-070097d0-5925-11e8-8d43-198a62d4b3f0.PNG)

I used the following to determine if the Permit is active:
WHERE 
	BP.SERV_PROV_CODE = 'MADISON' 
	AND BP.REC_STATUS = 'A'
	AND BP.B1_PER_GROUP = 'Permitting'
	AND BP.B1_PER_TYPE = 'Fire'
	AND BP.B1_PER_SUB_TYPE = 'Elevator'
	AND BP.B1_PER_CATEGORY = 'NA'
	AND BP.B1_APPL_STATUS NOT IN ('Closed', 'Inactive', 'Nonexist', 'Void')

Please contact me if you have any questions.

**Comment by RikiSjachrani-COM** [comment]
I'm just going to send a meeting invite to Jerry and include you and Jackie as well.

**Comment by RikiSjachrani-COM** [comment]
Sounds like the things we have to do are as follows, please let me know if I'm mistaken to missed something
 - [ ] Not on SS should stay assigned - No change in assignment here
 - [ ] Create new user in the elevator group named "UW Elevators"
 - [ ] Assign all the 413 elevators to this new user
 - [ ] Run the script in production
 - [ ] After the script, let's look at the Adhoc Reports under Elevator and modify the "All Elevators" to show all the elevators and the most recent inspection scheduled. Should match the number of active elevators

**Comment by ChrisHagen-COM** [comment]
I created a new elevator user named "FDUWELEVATORS" in dev/test/prod.

**Comment by BrendenPautsch-COM** [comment]
Riki,  Is Shannon Completely done?  As I was going through the process 1 more time, I thought of the few that were not on the spreadsheet, so I wrote something to update all the scheduled Annuals to Ollie (38) , however she has other Inspections scheduled (6), should I update those or just the Annuals?

**Comment by RikiSjachrani-COM** [comment]
Shannon is completed done, so I think it would be reasonable to assign any scheduled inspections over to Ollie

**Comment by RikiSjachrani-COM** [comment]
That doesn't sound right, I thought a annual inspection would be created on a successful inspection or reinspection. I can look back at specs... Jackie might remember as well

**Comment by BrendenPautsch-COM** [comment]
Riki,
I ran a query against the clean RptRepl dataset looking at the dates of the last Annual* Inspection and compared it the timing against todays date and the scheduled date in the provided spreadsheet.  I have noticed several inspection that have been missed, by a large range of months.  Can we contact Jerry and ask how we should handle those inspections.  See the follow Permits

- ELEV-2008-1133519 - 2nd Reinspect March 2018 - Not completed
- ELEV-2008-1149907 - Last Scheduled Inspection January 2016, not completed
- ELEV-2008-1003063 - January Annual Inspection Not Completed

**Comment by BrendenPautsch-COM** [comment]
Round 2:  After doing a deeper dive into the data and talking with Jerry, I have made adjustments to get the data as close as possible to the what was in the spreadsheet that Jerry provided.  I have attached the latest SQL with some better notation to better describe what my thoughts were while developing.  I requested a meeting with Riki for the this afternoon (5/22) to QA and run against the RptRepl database.

[SR-18-0000132.txt](https://github.com/City-of-Madison/Development-Team-2/files/2027319/SR-18-0000132.txt)
